magenta-lib/src/main/scala/magenta/tasks/UpdateCloudFormationTask.scala

```diff
@@ -1,18 +1,58 @@
 package magenta.tasks
 
 import com.amazonaws.AmazonServiceException
-import com.amazonaws.services.cloudformation.model.StackEvent
+import com.amazonaws.services.cloudformation.model.{AmazonCloudFormationException, StackEvent}
 import com.amazonaws.services.s3.AmazonS3
 import com.amazonaws.services.securitytoken.AWSSecurityTokenService
 import magenta.artifact.S3Path
 import magenta.deployment_type.CloudFormationDeploymentTypeParameters._
 import magenta.tasks.CloudFormation._
 import magenta.tasks.UpdateCloudFormationTask.CloudFormationStackLookupStrategy
-import magenta.{DeployReporter, DeployTarget, DeploymentPackage, KeyRing, Region, Stack, Stage}
+import magenta.{DeploymentPackage, DeployReporter, DeployTarget, KeyRing, Region, Stack, Stage}
 import org.joda.time.{DateTime, Duration}
 
 import scala.collection.convert.wrapAsScala._
 
+/**
+  * A simple trait to aid with attempting an update multiple times in the case that an update is already running.
+  */
+trait RetryCloudFormationUpdate {
```

Comment:
Superficially, this looks a lot like `RepeatedPollingCheck`. Is that a real similarity, or are there some fundamental differences?

Reply:
My initial version did use Repeated Polling check. But it can return a value and isn't actually a check. I concluded there was value in making a custom variant but could be convinced otherwise.